app/models/site_setting.rb

```diff
@@ -379,6 +382,34 @@ def self.clear_cache!(expire_theme_site_setting_cache: false)
     @blocked_attachment_filenames_regex = nil
     @allowed_unicode_username_regex = nil
   end
+
+  private
+
+  def extract_upload_ids_from_objects_value
+    upload_ids = []
+    return upload_ids if self.value.blank?
+
+    type_hash = SiteSetting.type_supervisor.type_hash(self.name)
+    return upload_ids unless type_hash[:schema]&.dig(:properties)
+
+    begin
+      parsed_value = JSON.parse(self.value)
+      parsed_value = [parsed_value] unless parsed_value.is_a?(Array)
+
+      parsed_value.each do |obj|
+        type_hash[:schema][:properties].each do |prop_key, prop_value|
+          next unless prop_value[:type] == "upload"
+
+          key = prop_key.to_s
+          upload_id = obj[key]
+          upload_ids << upload_id if upload_id.present?
+        end
+      end
+    rescue JSON::ParserError
+    end
+
+    upload_ids.compact.uniq
+  end
```

Comment:
I think you can use `SchemaSettingsObjectValidator#property_values_of_type` here instead of having to write custom code to manually parse out the value.